src/sparseml/pytorch/utils/quantization/helpers.py

```diff
@@ -64,6 +67,214 @@
+        quantization respectively
+    """
+
+    @staticmethod
+    def from_module(module: Module) -> "QATWrapper":
+        """
+        :param module: torch Module to create a QATWrapper for
+        :return: QATWrapper object created using the given Module as the forward
+            function. Will attempt to find any other named parameter of the QATWrapper
+            constructor from the attributes of the given Module
+        """
+        qat_wrapper_param_names = [
+            "num_inputs",
+            "num_outputs",
+            "input_qconfigs",
+            "output_qconfigs",
+        ]
+        qat_wrapper_kwargs = {
+            param_name: getattr(module, param_name)
+            for param_name in qat_wrapper_param_names
+            if hasattr(module, param_name)
+        }
+
+        return QATWrapper(forward_fn=module, **qat_wrapper_kwargs)
+
+    def __init__(
+        self,
+        forward_fn: Callable[[Any], Any],
+        num_inputs: int = 1,
+        num_outputs: int = 1,
+        input_qconfigs: Union[
+            "torch.quantization.QConfig", str, List["torch.quantization.QConfig"]
+        ] = "asymmetric",
+        output_qconfigs: Union[
+            "torch.quantization.QConfig", str, List["torch.quantization.QConfig"]
+        ] = "asymmetric",
+    ):
+        super().__init__()
+
+        if torch_quantization is None:
+            raise RuntimeError(
+                "Unable to import package torch.quantization. "
+                "Try upgrading your PyTorch version."
+            )
+
+        if not callable(forward_fn):
+            raise ValueError(
+                "forward_fn of QATWrapper must be callable. "
+                f"Received {type(forward_fn)}"
+            )
+
+        self.forward_fn = forward_fn
+        self.input_qconfigs = self._load_qconfigs(
+            "input_qconfigs", num_inputs, input_qconfigs
+        )
+        self.output_qconfigs = self._load_qconfigs(
+            "output_qconfigs", num_outputs, output_qconfigs
+        )
+
+        self.input_quant_stubs = torch.nn.ModuleList(
+            [torch_quantization.QuantStub() for _ in range(num_inputs)]
+        )
+        self.output_quant_stubs = torch.nn.ModuleList(
+            [torch_quantization.QuantStub() for _ in range(num_outputs)]
+        )
+        self.output_dequant_stubs = torch.nn.ModuleList(
+            [torch_quantization.DeQuantStub() for _ in range(num_outputs)]
+        )
+
+    def forward(self, *args, **kwargs) -> Any:
+        """
+        :param args: arguments to forward function; the first num_inputs of these args
+            will be wrapped by a QuantStub
+        :param kwargs: key word arguments to pass to the wrapped forward function
+        :return: outputs of the forward function with a QuantStub applied to the first
+            num_outputs outputs
+        """
+
+        qat_args = []
+
+        for idx, arg in enumerate(args):
```

Comment:
It's less of an edge case based on how most PyTorch code is implemented currently, but if someone does module(inp=tensor) we won't currently catch it. We should note this in the docs that only positional args are supported currently or try and update to be more generic so we can go through named kwargs by key and positional by index

Reply:
will add support for key word args